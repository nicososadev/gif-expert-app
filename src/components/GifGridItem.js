import React from 'react'
import propTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {
    
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propType = {
    title: propTypes.string.isRequired,
    url: propTypes.string.isRequired
}
