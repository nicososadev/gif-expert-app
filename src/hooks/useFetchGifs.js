import { useEffect, useState } from "react"
import { GetGifs } from "../helpers/GetGifs"

export const useFetchGifs = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true,
    })

    useEffect( () => {
        GetGifs( category )
            .then( images =>{
                setState({
                    data: images,
                    loading: false,
                })
            })
    }, [ category ])
    
    return state
}