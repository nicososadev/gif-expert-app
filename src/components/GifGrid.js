import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {

    const { data:images , loading } = useFetchGifs( category )
    
    return (
        <Fragment>
            <h3>{ category }</h3>

            { loading && <p className="card animate__animated animate__flash">Loading...</p> }

            <div className="card-grid">
            
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
                
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {

    category: PropTypes.string.isRequired

}
