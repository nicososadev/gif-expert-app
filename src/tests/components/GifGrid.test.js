import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente GifGrid', () => {
    
    test('Debe mostrar correctamente el Componente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        })
        
        const wrapper = shallow( <GifGrid category={ '' }/> )

        expect( wrapper ).toMatchSnapshot()
    })
    
    test('Debe mostar items cuando se cargan imagenes en useFetchGif', () => {
        
        const images = [
            {
                id: 'abc',
                title: 'titulo',
                url: 'https://url.com'
            }
        ]

        useFetchGifs.mockReturnValue({
            data: images,
            loading: false,
        })

        const wrapper = shallow( <GifGrid category={ '' }/> )

        const p = wrapper.find('p').exists()

        const componente = wrapper.find('GifGridItem').length

        expect( p ).toBe( false )

        expect( componente ).toBe( images.length )
    })
    
})
