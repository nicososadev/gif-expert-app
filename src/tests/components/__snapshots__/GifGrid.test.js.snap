// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe mostrar correctamente el Componente 1`] = `
<Fragment>
  <h3 />
  <p
    className="card animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
